### ISSUE TITLE
Narrator table shortcut Ctr+Alt+Arrow not working on tables

### ISSUE THREAD

**Issue by clmori** [microsoft/fluentui#11243]
### Environment Information

- **Package version(s):** 7.60.1
- **Browser and OS versions:** Microsoft Edge 44.18362.449.0

### Describe the issue:

Narrator's shortcut "Ctr+Alt+Arrow" keys to navigate through table elements only works for header, and is not working for items cells.

### Please provide a reproduction of the issue in a codepen:
You can see this issue in the fabric DetailsList - Grouped page linked below:
https://developer.microsoft.com/en-us/fabric#/controls/web/detailslist/grouped

#### Actual behavior:

When narrator is used and "Ctr+Alt+Arrow" is used, nothing happens

#### Expected behavior:

When narrator is used and "Ctr+Alt+Arrow" is used, user is able to navigate through the table

**Comment by rkalivarapu** [comment]
We are able to navigate using Table keyboard shortcuts (Ctrl+Alt+Arrow Keys) only on header columns/cells, but not on item cells because of the groups/group headers.

**Comment by aneeshack4** [comment]
This does not repo when scan mode is off so this is a scan mode issue. Please contact the Narrator team.

**Comment by smhigley** [comment]
@aneeshack4 this isn't actually a Narrator issue, the control in question is not using a valid semantic structure, and that is why Narrator (and NVDA, and JAWS) fail to navigate it properly. Here's a screenshot of the accessibility tree of the grouped detailslist on this page: https://developer.microsoft.com/en-us/fabric#/controls/web/detailslist/grouped

![Firefox accessibility tree of detailslist table](https://user-images.githubusercontent.com/3819570/69199774-e42e7c00-0aed-11ea-8a27-44c9679b8a8f.png)

In the accessibility tree, there are both elements contained within the parent table that are not rows, cells, or rowgroups, and also `table` elements that are direct children of the parent `table`, instead of children of a cell within that table.

Hope that makes sense! I can add more detail if needed.

**Comment by clmori** [comment]
@aneeshack4 let us know if this provides enough info

**Comment by aneeshack4** [comment]
@smhigley Thank you for providing additional detail. Is there a source that you're using to see the valid semantic structure? I ask this because the CTRL + ALT + ARROW short cut does work when Narrator's not in scan mode.

**Comment by smhigley** [comment]
@aneeshack4 I'm using Firefox's accessibility tab in devtools to inspect the accessibility tree.

Re: Narrator table commands: these are scan mode interactions, it doesn't really make sense to say it works outside of scan mode, because then you're not using Narrator's table commands (same also goes for JAWS and NVDA).

**Comment by aneeshack4** [comment]
@smhigley I meant is there a source where the correct semantic structure for tables/details lists has been stated like the aria spec? And to clarify, Narrator table commands like CTRL+ALT+ARROW are only valid during scan mode? Is there documentation saying this too? We've had a lot of issues with scan mode in Narrator where the resolution most of the time depends on added support from Narrator for that mode which is why I tested outside of scan mode also.

**Comment by aneeshack4** [comment]
We see that this repros but like I said before, only in Narrator scan mode. Scan mode isn't specified in the ARIA spec and therefore is not a blocking issue. And scan mode support needs to be added by the Narrator team which we are waiting on for other components' issues too.

**Comment by smhigley** [comment]
@aneeshack4 the aria spec says you need to respect the host language's semantics, including the content model for analogous HTML elements. The content model for `<table>` is defined here: https://html.spec.whatwg.org/multipage/tables.html#the-table-element. A `<table>` element may only contain elements with `row` and `rowgroup` semantics, not other `table` elements. You can also go through the ARIA spec and look at "Required Owned Elements" and "Required Context Role" for each of the table roles.

Re: Narrator scan mode. The particular behavior of narrator (and other screen readers) isn't something you need to directly worry about, and isn't defined in spec. So long as you define semantically correct tables, Narrator, JAWS, and NVDA will handle presenting that tabular data to users in the way those users expect. This bug isn't about Narrator's scan mode failing, this is about not defining semantically correct tables, and a side effect of that is that screen readers are not able to parse them.

Hope that helps! Let me know if you have any other questions.

**Comment by aneeshack4** [comment]
Adding this to the backlog for DetailsList.

**Comment by XianghanWang** [comment]
Any update on this?

**Comment by wiyi123** [comment]
Is there any update on this? Our team has an open accessibility bug for this.

**Comment by paulgildea** [comment]
Pulling this back to the top as there appears some product facing issues associated with this.

**Comment by smhigley** [comment]
@paulgildea would you mind if I took this one on? I think I might be able to fix some of the table structure without altering the DOM, though there is a bit of an issue around the markup for the row checkbox and expand/collapse button not being contained in a gridcell.

**Comment by xugao** [comment]
@smhigley - before you start the fix, pls confirm your approach is okay with @cschlechty  (there is always a lot risk involves fixing details list bugs, he is the best person to help)
